src/App.tsx

```diff
@@ -1,61 +1,132 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { useState } from 'react';
+import { TodoListType, TodosType, User } from './api/type/type';
+
+function findUserById(id: number, userList: User[]): User | undefined {
+  return userList.find(user => id === user.id);
+}
+
+function makeTodoList(todos: TodosType[], userList: User[]): TodoListType[] {
+  return todos.map(todo => {
+    return {
+      ...todo,
+      user: findUserById(todo.userId, userList),
+    };
+  });
+}
 
 export const App = () => {
+  const [todoList, setTodoList] = useState(
+    makeTodoList(todosFromServer, usersFromServer),
+  );
+  const [selectedUser, setSelectedUser] = useState<number>(0);
+  const [title, setTitle] = useState<string>('');
+  const [errorSelectedUser, setErrorSelectedUser] = useState(false);
+  const [errorTitle, setErrorTitle] = useState(false);
+
+  function handleSubmit(): void {
+    if (selectedUser === 0) {
+      setErrorSelectedUser(true);
+    }
+
+    if (title.trim().length === 0) {
+      setErrorTitle(true);
+    }
+
+    if (selectedUser !== 0 && title.trim().length > 0) {
+      const newTodo: TodoListType = {
+        id: Math.max(0, ...todosFromServer.map(a => a.id)) + 1, // Ensure id is valid
+        title: title.trim(),
+        completed: false,
+        userId: selectedUser,
+        user: findUserById(selectedUser, usersFromServer),
+      };
+
+      setTodoList(curTodolist => [...curTodolist, newTodo]);
+      setSelectedUser(0);
+      setTitle('');
+      setErrorSelectedUser(false);
+      setErrorTitle(false);
+    }
+
+    return undefined;
```

Comment:
The `return undefined;` statement is unnecessary in this context. The function `handleSubmit` does not need to return anything, so you can safely remove this line.